pythonfmu/pythonfmu-export/src/pythonfmu/PySlaveInstance.cpp

```diff
@@ -25,11 +25,69 @@ inline std::string getline(const std::string& fileName)
     return line;
 }
 
+
+std::string searchLeafClassName(PyObject* pLocals)
+{
+    std::string deepestFile = "";
+    Py_ssize_t deepestChain = 0;
+    PyObject* key, * value;
+    Py_ssize_t pos = 0;
+
+    while (PyDict_Next(pLocals, &pos, &key, &value)) {
+        // Check if element in namespace is a class
+        if (!PyType_Check(value)) {
+            continue;
+        }
+
+        PyObject* pMroAttribute = PyObject_GetAttrString(value, "__mro__");
+
+        if (pMroAttribute != NULL && PySequence_Check(pMroAttribute)) {
+            std::regex pattern("<class '[^']+\\.([^']+)'");
+            PyObject* pMROList = PySequence_List(pMroAttribute);
+
+            for (Py_ssize_t i = 0; i < PyList_Size(pMROList); ++i) {
+                PyObject* pItem = PyList_GetItem(pMROList, i);
+                std::smatch match;
+                const char* className = PyBytes_AsString(PyUnicode_AsUTF8String(PyObject_Repr(pItem)));
+
+                std::string str(className);
+                bool isMatch = std::regex_search(str, match, pattern);
+
+                // If regex match is successfull, and found Fmi2Slave at the deepest level then update state
+                if (isMatch && i > deepestChain && match[1] == "Fmi2Slave") {
+                    deepestFile = PyBytes_AsString(PyUnicode_AsUTF8String(key));
+                    deepestChain = i;
+                }
+            }
+        }
+        Py_DECREF(pMroAttribute);
+    }
+    return deepestFile;
+}
+
+PyObject* findClassFromBinary(const std::string& resources, const std::string& moduleName) {
+
+    PyObject* pyModule = PyImport_ImportModule(moduleName.c_str());
+    if (pyModule == nullptr) {
+        return nullptr;
+    }
+
+    PyObject* pGlobals = PyModule_GetDict(pyModule);
+    auto deepestFile = searchLeafClassName(pGlobals);
```

Comment:
@Jorgelmh passing `pGlobals` here

Reply:
Inheritance seems to work fine, just tried using the new binaries in component-model and seems to work as expected. I'd still recommend we wait until Lars reviews this part as I'm not sure if it might affect other functionality.